src/types/memoryview.rs

```diff
@@ -0,0 +1,123 @@
+use std::marker::PhantomData;
+use std::os::raw::c_char;
+
+use crate::{AsPyPointer, PyAny, PyResult, Python};
+use crate::buffer::{Element, PyBuffer};
+use crate::ffi::{self, Py_ssize_t};
+use crate::types::PySequence;
+
+/// Represents a Python `memoryview`.
+#[repr(transparent)]
+pub struct PyMemoryView<T>(PyAny, PhantomData<T>);
```

Comment:
The choice to make this generic is interesting, and I've been thinking about this further. At the moment none of PyO3's types (other than buffer) are generic. Also `PyMemoryView_Check` won't check the generic type is correct, which makes it easy for this type to hold memory views with the wrong T.

For these reasons I think it's better if we remove the generic type from this and just have `PyMemoryView`.

Reply:
Done